hazelcast-spring/src/main/java/com/hazelcast/spring/HazelcastConfigBeanDefinitionParser.java

```diff
@@ -833,9 +833,55 @@ public void handleWanReplication(Node node) {
                         }
                     }
                     targetClusters.add(childBeanDefinition);
+                } else if ("wan-publisher".equals(nName)) {
+                    final BeanDefinitionBuilder publisherConfigBuilder = createBeanBuilder(WanPublisherConfig.class);
+                    final AbstractBeanDefinition childBeanDefinition = publisherConfigBuilder.getBeanDefinition();
+                    fillAttributeValues(n, publisherConfigBuilder, Collections.<String>emptyList());
+
+                    final NamedNodeMap attrs = n.getAttributes();
+                    Node classNameNode = attrs.getNamedItem("class-name");
+                    String className = classNameNode != null ? getTextContent(classNameNode) : null;
+                    Node implNode = attrs.getNamedItem("implementation");
+                    String implementation = implNode != null ? getTextContent(implNode) : null;
+                    publisherConfigBuilder.addPropertyValue("className", className);
+                    if (implementation != null) {
+                        publisherConfigBuilder.addPropertyReference("implementation", implementation);
+                    }
+                    Assert.isTrue(className != null || implementation != null, "One of 'class-name' or 'implementation' "
```

Comment:
what is the expected behaviour when both of the `className` and `implementation` are specified?